From 4416ffd72d3c4b39ec2494b5f60d83a2fc482940 Mon Sep 17 00:00:00 2001
From: Varun Berry <vberry@google.com>
Date: Fri, 5 Nov 2021 10:56:17 -0700
Subject: [PATCH] Use an explicit intent to start
 VoicemailChangePinActivityPeer.

This fixes an issue where the CHANGE_PIN intent can be intercepted.

Bug: 186405602
Test: manual.
Merged-In: If31e733c4f94fa4d0793a7538ec76a10752329b6
Change-Id: I0ec5921f2723430d16d6f101ba0ad351ab0cf73c
(cherry picked from commit 02e1666deb3f1f638a0ca5510e75b126c13ddbf1)
Merged-In:I0ec5921f2723430d16d6f101ba0ad351ab0cf73c
---
 .../dialer/voicemail/listui/error/VoicemailErrorMessage.java  | 2 +-
 .../com/android/dialer/voicemail/settings/AndroidManifest.xml | 4 ----
 .../dialer/voicemail/settings/VoicemailChangePinActivity.java | 1 -
 3 files changed, 1 insertion(+), 6 deletions(-)

diff --git a/java/com/android/dialer/voicemail/listui/error/VoicemailErrorMessage.java b/java/com/android/dialer/voicemail/listui/error/VoicemailErrorMessage.java
index 14d6cffdc..a3c294d66 100644
--- a/java/com/android/dialer/voicemail/listui/error/VoicemailErrorMessage.java
+++ b/java/com/android/dialer/voicemail/listui/error/VoicemailErrorMessage.java
@@ -150,7 +150,7 @@ public class VoicemailErrorMessage {
           public void onClick(View v) {
             Logger.get(context)
                 .logImpression(DialerImpression.Type.VOICEMAIL_ALERT_SET_PIN_CLICKED);
-            Intent intent = new Intent(VoicemailChangePinActivity.ACTION_CHANGE_PIN);
+            Intent intent = new Intent(context, VoicemailChangePinActivity.class);
             intent.putExtra(VoicemailClient.PARAM_PHONE_ACCOUNT_HANDLE, phoneAccountHandle);
             context.startActivity(intent);
           }
diff --git a/java/com/android/dialer/voicemail/settings/AndroidManifest.xml b/java/com/android/dialer/voicemail/settings/AndroidManifest.xml
index 8506ddd41..f25858e9d 100644
--- a/java/com/android/dialer/voicemail/settings/AndroidManifest.xml
+++ b/java/com/android/dialer/voicemail/settings/AndroidManifest.xml
@@ -29,10 +29,6 @@
         android:parentActivityName="com.android.dialer.app.settings.DialerSettingsActivity"
         android:theme="@style/SettingsStyle"
         android:windowSoftInputMode="stateVisible|adjustResize">
-      <intent-filter>
-        <action android:name="com.android.dialer.action.CHANGE_PIN"/>
-        <category android:name="android.intent.category.DEFAULT"/>
-      </intent-filter>
     </activity>
     <activity
         android:name=".RecordVoicemailGreetingActivity"
diff --git a/java/com/android/dialer/voicemail/settings/VoicemailChangePinActivity.java b/java/com/android/dialer/voicemail/settings/VoicemailChangePinActivity.java
index 4e22fb335..ebebdf7d4 100644
--- a/java/com/android/dialer/voicemail/settings/VoicemailChangePinActivity.java
+++ b/java/com/android/dialer/voicemail/settings/VoicemailChangePinActivity.java
@@ -67,7 +67,6 @@ public class VoicemailChangePinActivity extends Activity
     implements OnClickListener, OnEditorActionListener, TextWatcher {
 
   private static final String TAG = "VmChangePinActivity";
-  public static final String ACTION_CHANGE_PIN = "com.android.dialer.action.CHANGE_PIN";
 
   private static final int MESSAGE_HANDLE_RESULT = 1;
 
-- 
2.33.1

